src/Microsoft.Net.Http.Headers/MediaTypeHeaderValue.cs

```diff
@@ -441,12 +441,24 @@ private static int GetMediaTypeLength(string input, int startIndex, out MediaTyp
             return current - startIndex;
         }
 
-        private static int GetMediaTypeExpressionLength(string input, int startIndex, out string mediaType)
+        internal static int ParseMediaTypeExpresion<T, K>(
+            string input,
+            int startIndex,
+            out K mediaType,
+            Func<string, int, int, T> typeAndSubtypeFactory,
+            Func<T, T, K> typeAndSubtypeCombinator)
```

Comment:
Waz a "combinator"? Perhaps `typeAndSubtypeCombiner`? Same for `nameAndValueCombinator` later.